ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -507,84 +464,6 @@ class ClientQueryViewController: UITableViewController, Themeable {
-				return (false, "File name cannot contain / or \\")
-			} else {
-				return (true, nil)
-			}
-		}, completion: { newName, _ in
-
-			guard newName != nil else {
-				return
-			}
-
-			if let progress = self.core.move(item, to: self.query.rootItem, withName: newName!, options: nil, resultHandler: { (error, _, _, _) in
-				if error != nil {
-					Log.log("Error \(String(describing: error)) renaming \(String(describing: item.path))")
-
-					completionHandler?(false)
-				} else {
-					completionHandler?(true)
-				}
-			}) {
-				self.progressSummarizer?.startTracking(progress: progress)
-			}
-		})
-
-		renameViewController.navigationItem.title = "Rename".localized
-
-		let navigationController = ThemeNavigationController(rootViewController: renameViewController)
-		navigationController.modalPresentationStyle = .overFullScreen
-
-		self.present(navigationController, animated: true, completion: viewDidAppearHandler)
-	}
-
-	func delete(_ item: OCItem, viewDidAppearHandler: ClientActionVieDidAppearHandler? = nil, completionHandler: ClientActionCompletionHandler? = nil) {
-		let alertController = UIAlertController(
-			with: item.name!,
-			message: "Are you sure you want to delete this item from the server?".localized,
-			destructiveLabel: "Delete".localized,
-			preferredStyle: UIDevice.current.isIpad() ? UIAlertControllerStyle.alert : UIAlertControllerStyle.actionSheet,
-			destructiveAction: {
-				if let progress = self.core.delete(item, requireMatch: true, resultHandler: { (error, _, _, _) in
-					if error != nil {
-						Log.log("Error \(String(describing: error)) deleting \(String(describing: item.path))")
-
-						completionHandler?(false)
-					} else {
-						completionHandler?(true)
-					}
-				}) {
-					self.progressSummarizer?.startTracking(progress: progress)
-				}
-			}
-		)
-
-		self.present(alertController, animated: true, completion: viewDidAppearHandler)
-	}
-
-	func move(_ item: OCItem, viewDidAppearHandler: ClientActionVieDidAppearHandler? = nil, completionHandler: ClientActionCompletionHandler? = nil) {
-		let directoryPickerVC = ClientDirectoryPickerViewController(core: self.core, path: "/", completion: { (selectedDirectory) in
-
-			if let progress = self.core.move(item, to: selectedDirectory, withName: item.name, options: nil, resultHandler: { (error, _, _, _) in
-				if error != nil {
-					Log.log("Error \(String(describing: error)) moving \(String(describing: item.path))")
-					completionHandler?(false)
-				} else {
-					completionHandler?(true)
-				}
-			}) {
-				self.progressSummarizer?.startTracking(progress: progress)
-			}
-		})
-
-		let pickerNavigationController = ThemeNavigationController(rootViewController: directoryPickerVC)
-		self.navigationController?.present(pickerNavigationController, animated: true)
-	}
-
 	func createFolder(viewDidAppearHandler: ClientActionVieDidAppearHandler? = nil, completionHandler: ClientActionCompletionHandler? = nil) {
```

Comment:
"Create folder" should ultimately also move to an `Action` subclass. To make it possible to be found, we'd need to add a new location identifier:

```
extension OCExtensionLocationIdentifier {
	static let tableRow: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier("tableRow") //!< Present as table row action
	static let moreItem: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier("moreItem") //!< Present in "more" card view for a single item
	static let moreFolder: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier("moreFolder") //!< Present in "more" options for a whole folder
	static let toolbar: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier("toolbar") //!< Present in a toolbar
	// NEW:
	static let sortBar: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier("sortBar") //!< Present in the sort bar on top of file lists
}
```

The `CreateFolderAction.locations` could then return `[.moreFolder, .sortBar]`, so it's only available when tapping the three dots in the navigation bar (in the future) or the "New folder" icon in the sort bar.